deps/exokit-bindings/webglcontext/src/webgl.cc

```diff
@@ -4938,12 +4997,33 @@ NAN_METHOD(WebGLRenderingContext::GetSupportedExtensions) {
   // glGetIntegerv(GL_NUM_EXTENSIONS, &numExtensions);
 
   int numExtensions = sizeof(webglExtensions)/sizeof(webglExtensions[0]);
-  Local<Array> result = Nan::New<Array>(numExtensions);
+  Local<Array> result = Nan::New<Array>(0);
+
+  GLint n=0;
+  glGetIntegerv(GL_NUM_EXTENSIONS, &n);
+
+  for (GLint i=0; i<n; i++)
+  {
+    const char* extension = (const char*)glGetStringi(GL_EXTENSIONS, i);
+
+    for (GLint j = 0; j < numExtensions; j++) {
+      bool matchesSupportedExtension = (strcmp(extension+3,webglExtensions[j]) == 0);
+      if(matchesSupportedExtension){
+        bool hasWebGL = (strncmp(webglExtensions[j], "WEBGL", 5) == 0);
+          if(!hasWebGL) {
+            int numResults = result->Length();
+            result->Set(numResults, JS_STR(webglExtensions[j]));
+          }
+      }
+    }
+  }
   for (GLint i = 0; i < numExtensions; i++) {
-    // char *extension = (char *)glGetStringi(GL_EXTENSIONS, i);
-    result->Set(i, JS_STR(webglExtensions[i]));
+    bool hasWebGL = (strncmp(webglExtensions[i], "WEBGL", 5) == 0);
```

Comment:
If the extension was already included in the above list, it would be duplicated here, no?

Reply:
Actually, never mind, the above list generation checks for non-webgl. 👍